### PR TITLE
Origin/somkape/code flow

### PR DESCRIPTION
**Pull request by sangonzal**
GitHub does not let you commit to another person's PR so I made all necessary changes based on feedback from Petro's PR into this one.

### CODE REVIEW

File: src/main/java/com/microsoft/aad/adal4j/AuthenticationException.java [hunk 1]
```diff
@@ -30,35 +30,44 @@
...
     /**
      * Constructor
-     * 
-     * @param message
-     *            string error message
-     * @param t
-     *            Throwable object
+     *
+     * @param message string error message
+     * @param t       Throwable object
```
Comment (henrik-me) [inline src/main/java/com/microsoft/aad/adal4j/AuthenticationException.java]: >        [](start = 15, length = 7)

indentation?

Comment (henrik-me) [inline src/main/java/com/microsoft/aad/adal4j/AuthenticationException.java]: >t [](start = 14, length = 1)

t ?

**Comment by henrik-me** [comment]
Approving though there are a few comments worth addressing.

File: src/main/java/com/microsoft/aad/adal4j/ClientDataHttpHeaders.java [hunk 2]
```diff
@@ -1,125 +1,126 @@
...
+
+    public final static String CORRELATION_ID_HEADER_NAME = "client-request-id";
+    public final String correlationIdHeaderValue;
+
+    public final static String REQUEST_CORRELATION_ID_IN_RESPONSE_HEADER_NAME = "return-client-request-id";
+    public final static String REQUEST_CORRELATION_ID_IN_RESPONSE_HEADER_VALUE = "true";
+    private final String headerValues;
+    private final Map<String, String> headerMap = new HashMap<String, String>();
+
+    public static String generateCorrelationId(){
```
[the remark this answers is not in the document]
Comment (sangonzal) [reply src/main/java/com/microsoft/aad/adal4j/ClientDataHttpHeaders.java]: You're right, it is not used anywhere else. I have changed the scope to private

File: src/main/java/com/microsoft/aad/adal4j/AuthenticationException.java (hunk 1 above)
Comment (sangonzal) [reply src/main/java/com/microsoft/aad/adal4j/AuthenticationException.java]: Fixed

**Pull request merged**